pkg/artifactory/resource/configuration/resource_artifactory_proxy.go

```diff
@@ -0,0 +1,267 @@
+		_, err := m.(*resty.Client).R().SetResult(&proxiesConfig).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+
+		// Unpacking HCL to compare the names of the property sets with the XML data we will get from the API
+		unpackedProxy := unpackProxy(d)
+		matchedProxyConfig := FindConfigurationById[Proxy](proxiesConfig.Proxies, unpackedProxy.Key)
+		if matchedProxyConfig == nil {
+			return nil
+		}
+
+		return packProxy(matchedProxyConfig, d)
+	}
+
+	var resourceProxyUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedProxy := unpackProxy(d)
+
+		///* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		//There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		//GET call structure has "propertySets -> propertySet -> Array of property sets".
+		//PATCH call structure has "propertySets -> propertySet (dynamic sting). Property name and predefinedValues names are also dynamic strings".
+		//Following nested map of string structs are constructed to match the usage of PATCH call with the consideration of dynamic strings.
+		//*/
+		var body = map[string]map[string]Proxy{
+			"proxies": {
+				unpackedProxy.Key: unpackedProxy,
+			},
+		}
+
+		content, err := yaml.Marshal(&body)
+		if err != nil {
+			return diag.Errorf("failed to marshal proxy during Update")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		d.SetId(unpackedProxy.Key)
+		return resourceProxyRead(ctx, d, m)
+	}
+
+	var resourceProxyDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		proxiesConfig := &Proxies{}
+
+		response, err := m.(*resty.Client).R().SetResult(&proxiesConfig).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		if response.IsError() {
+			return diag.Errorf("got error response for API: /artifactory/api/system/configuration request during Read")
+		}
+
+		unpackedProxy := unpackProxy(d)
+		matchedProxyConfig := FindConfigurationById[Proxy](proxiesConfig.Proxies, unpackedProxy.Key)
+		if matchedProxyConfig == nil {
+			return nil
+		}
+
+		var body = map[string]map[string]string{
+			"proxies": {
+				matchedProxyConfig.Key: "~",
+			},
+		}
+
+		content, err := yaml.Marshal(&body)
+		if err != nil {
+			return diag.Errorf("failed to marshal proxy during Delete")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Delete")
+		}
+
+		return nil
+	}
+
+	var verifyCrossDependentValues = func(ctx context.Context, diff *schema.ResourceDiff, v interface{}) error {
```

Comment:
Here `ctx` may not be giving compilation error because it's part of the function definition, but do you think you need this? I don't see this being used in the function body.

Reply:
Good catch, will remove it in future PR.